sct-commons/src/main/java/org/lfenergy/compas/sct/commons/ExtRefEditorService.java

```diff
@@ -78,6 +79,37 @@ private static List<TIED> getIedSources(SclRootAdapter sclRootAdapter, TCompasBa
                 .toList();
     }
 
+    /**
+     * Provides a list of ExtRef and associated Bay <br/>
+     * - The location of ExtRef should be in LDevice (inst=LDEPF) <br/>
+     * - ExtRef that lacks Bay or ICDHeader Private is not returned <br/>
+     *
+     * @param sclReportItems List of SclReportItem
+     * @return list of ExtRef and associated Bay
+     */
+    private List<ExtRefInfo.ExtRefWithBayReference> getExtRefWithBayReferenceInLDEPF(final LDeviceAdapter lDeviceAdapter, final List<SclReportItem> sclReportItems) {
```

Comment:
Can you remove the use of adapters ?  
```java
private List<ExtRefInfo.ExtRefWithBayReference> getExtRefWithBayReferenceInLDEPF(TDataTypeTemplates dataTypeTemplates, TIED tied, final TLDevice tlDevice, final List<SclReportItem> sclReportItems) {
        List<ExtRefInfo.ExtRefWithBayReference> extRefBayReferenceList = new ArrayList<>();
        String ldevicePath = "SCL/IED/AccessPoint/Server/LDevice[" + tlDevice.getInst() + "]";
        Optional<TCompasBay> tCompasBay = PrivateUtils.extractCompasPrivate(tied, TCompasBay.class);
        if (tCompasBay.isEmpty()) {
            sclReportItems.add(SclReportItem.error(ldevicePath, "The IED has no Private Bay"));
            if (PrivateUtils.extractCompasPrivate(tied, TCompasICDHeader.class).isEmpty()) {
                sclReportItems.add(SclReportItem.error(ldevicePath, "The IED has no Private compas:ICDHeader"));
            }
            return Collections.emptyList();
        }

        if (dataTypeTemplatesService.isDoModAndDaStValExist(dataTypeTemplates, tlDevice.getLN0().getLnType())) {
            extRefBayReferenceList.addAll(tlDevice.getLN0()
                    .getInputs()
                    .getExtRef().stream()
                    .map(extRef -> new ExtRefInfo.ExtRefWithBayReference(tied.getName(), tCompasBay.get(), extRef)).toList());
        } else {
            sclReportItems.add(SclReportItem.error(ldevicePath, "DO@name=Mod/DA@name=stVal not found in DataTypeTemplate"));
        }
        return extRefBayReferenceList;
    }
```
and :  
```java
@Override
    public List<SclReportItem> manageBindingForLDEPF(SCL scd, EPF epf) {
        List<SclReportItem> sclReportItems = new ArrayList<>();
        SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
        if (!epf.isSetChannels()) return sclReportItems;
        new IedService()
                .getFilteredIeds(scd, ied -> !ied.getName().contains("TEST"))
                .forEach(tied -> ldeviceService.findLdevice(tied, tlDevice -> LDEVICE_LDEPF.equals(tlDevice.getInst()))
                        .ifPresent(tlDevice -> getExtRefWithBayReferenceInLDEPF(scd.getDataTypeTemplates(), tied, tlDevice, sclReportItems)
                                .forEach(extRefBayRef -> epf.getChannels().getChannel().stream().filter(tChannel -> doesExtRefMatchLDEPFChannel(extRefBayRef.extRef(), tChannel))
                                        .findFirst().ifPresent(channel -> {
                                            List<TIED> iedSources = getIedSources(sclRootAdapter, extRefBayRef.compasBay(), channel);
                                            if (iedSources.size() == 1) {
                                                updateLDEPFExtRefBinding(extRefBayRef.extRef(), iedSources.get(0), channel);
                                                sclReportItems.addAll(updateLDEPFDos(new LDeviceAdapter(new IEDAdapter(sclRootAdapter), tlDevice), extRefBayRef.extRef(), channel));
                                            } else {
                                                if (iedSources.size() > 1) {
                                                    sclReportItems.add(SclReportItem.warning(null, "There is more than one IED source to bind the signal " +
                                                            "/IED@name=" + extRefBayRef.iedName() + "/LDevice@inst=LDEPF/LN0" +
                                                            "/ExtRef@desc=" + extRefBayRef.extRef().getDesc()));
                                                }
                                                // If the source IED is not found, there will be no update or report message.
                                            }
                                        }))));
        return sclReportItems;
    }
```

Reply:
OK